src/hpm.c

```diff
@@ -1,98 +1,153 @@
-        printf("Algo Malio Sal. Reintente\r\n");
-        return;
+        medida=false;
     }
-    else if ((carga[0]==0x40)&&(carga[1]==0x05)&&(carga[2]==0x04))
-    {
-        //Mostrar mediciones
-        printf(" PM 2.5: %d, PM 10: %d\r\n",((carga[3]<<8)+carga[4]),((carga[5]<<8)+carga[6]));
+    return medida;
+}
+
+uint16_t getLastPM25 (void)
+{
+    /*
+    //Quizas no sea necesario este bloque de codigo, teniendo en cuenta la descripcion
+    //de la funcion HPMinput() - TODO: Eliminar este comentario, de ser necesario
+    if (autosend == true){
+        pm25=(carga[6]<<8)+carga[7];
     }
-    else if ((carga[0]==0xA5)&&(carga[1]==0xA5))
-    {
-        printf(" OK\r\n");
-        /*
-        if(flag == 1)
-        {
-                printf(" Medicion Iniciada\r\n");
-        }
-        else{
-            printf(" Medicion Detenida\r\n");
-        }
-        // */
-    } else {
-        //Caso por defecto, procesamiento no implementado.
-        printf("Respuesta (%i):",peso);
-        for (unsigned char i=0;i<peso;i++) {
-            printf(" %02X",carga[i]);
-        }
-        printf("\n\r");
+    else{
+        hpmSendReadMeasure();
+        pm25=(carga[3]<<8)+carga[4];
     }
+    // */
+    return pm25;
 }
 
-void smHPM (char val) {
-    if (hpm.bandera) {
-        //Esta corriendo, quiere decir que aun no se ha cortado la recepcion
-        iniciatim();
-        hpm.buffer[hpm.idx++] = val;
-    } else {
-        //Desborde, por tanto: fin de mensaje
-        
+uint16_t getLastPM10(void)
+{
+    /*
+    //Quizas no sea necesario este bloque de codigo, teniendo en cuenta la descripcion
+    //de la funcion HPMinput() - TODO: Eliminar este comentario, de ser necesario
+    if (autosend == true){
+        pm10=(carga[8]<<8)+carga[9];
+    }
+    else{
+        hpmSendReadMeasure();
+        pm10=(carga[5]<<8)+carga[6];
     }
+    // */
+    return pm10;
+}
+void HPMinput(char octeto) {
+    if (rxIdx < RXBUFLEN) {
+        rxBuffer[rxIdx++] = octeto;
+        if (autosend) {
+            if (rxIdx == RXBUFLEN){
+                rxBuffer[rxIdx++]=octeto;
+                pm10=(rxBuffer[8]<<8)+rxBuffer[9];
+                pm25=(rxBuffer[6]<<8)+rxBuffer[7];
+                rxIdx=0;
+                updated=true;//Verificar
+            }else{
+                switch(ultimoComandoEnviado){
+                    case ReadMeasure:
+                        if (rxIdx == RXBUFLEN){
+                           rxBuffer[rxIdx++]=octeto;
+                           pm10=(rxBuffer[5]<<8)+rxBuffer[6];
+                           pm25=(rxBuffer[3]<<8)+rxBuffer[4];
+                           rxIdx=0;
+                           updated=true;//Verificar
+                           break;
+                     }
+                    
+                } 
+            }  
+        } else {
+            /*TODO: Procesamiento segun Tabla 5 del Datasheet*/
+        }
+    }
+}
+         
+void enviaOrden (char *const orden, char largo) {
```

Comment:
El prototipo de la función también tiene `static`

```suggestion
static void enviaOrden (char *const orden, char largo) {
```